sdk/typescript/src/signers/signer-with-provider.ts

```diff
@@ -330,7 +332,7 @@ export abstract class SignerWithProvider implements Signer {
    * @returns total gas cost estimation
    * @throws whens fails to estimate the gas cost
    */
-  async getGasCostEstimation(tx: SignableTransaction | string) {
+  async getGasCostEstimation(tx: Parameters<SignerWithProvider['dryRunTransaction']>['0']) {
```

Comment:
Could also just collect all args and pass them all through so that you don't need to keep these functions in sync.